fe/fe-core/src/main/java/com/starrocks/external/hive/NewHivePartitionName.java

```diff
@@ -0,0 +1,116 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
+
+package com.starrocks.external.hive;
+
+import com.starrocks.analysis.BoolLiteral;
+import com.starrocks.analysis.LiteralExpr;
+import com.starrocks.analysis.NullLiteral;
+import com.starrocks.catalog.NullablePartitionKey;
+import com.starrocks.catalog.PartitionKey;
+import com.starrocks.external.Utils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+// TODO(stephen): rename NewHivePartitionName to HivePartitionName and remove original HivePartitionName
+//  After refactoring others related codes
+public class NewHivePartitionName {
+    private final String databaseName;
+    private final String tableName;
+    private final List<String> partitionValues;
+
+    // partition name eg: "year=2020/month=10/day=10"
+    private final Optional<String> partitionNames;
+
+    public NewHivePartitionName(String dbName, String tableName, List<String> partitionValues) {
+        this(dbName, tableName, partitionValues, Optional.empty());
+    }
+
+    public NewHivePartitionName(String databaseName,
+                                String tableName,
+                                List<String> partitionValues,
+                                Optional<String> partitionNames) {
+        this.databaseName = databaseName;
+        this.tableName = tableName;
+        this.partitionValues = partitionValues;
+        this.partitionNames = partitionNames;
+    }
+
+    public static NewHivePartitionName of(String dbName, String tblName, List<String> partitionValues) {
+        return new NewHivePartitionName(dbName, tblName, partitionValues);
+    }
+
+    public static NewHivePartitionName of(String dbName, String tblName, String partitionNames) {
+        return new NewHivePartitionName(dbName, tblName, Utils.toPartitionValues(partitionNames), Optional.of(partitionNames));
+    }
+
+    public static List<String> fromPartitionKey(PartitionKey key) {
```

Comment:
the retrun type of `NewHivePartitionName::fromPartitionKey` is not a `NewHivePartitionName`, seems a little confused

Reply:
I'll remove it to other class in next patch